ansible_collections/arista/avd/roles/eos_config_deploy_cvp/docs/avd-to-cvaas.md

```diff
@@ -0,0 +1,169 @@
+
+## Testing connectivity and authentication between AVD and CVaaS
+
+```text
+root@6e3d94f50dca:/workspace# ansible-playbook playbooks/cvaas_facts.yml
+
+PLAY [Playbook to demonstrate cv_container module.] *********************************************************************************************************************************
+
+TASK [Gather CVaaS facts from cvaas] ************************************************************************************************************************************************
+ok: [cvaas]
+
+PLAY RECAP **************************************************************************************************************************************************************************
+cvaas                      : ok=1    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0
+```
+
+## Sample playbook cvaas_facts.yml
+
+```text
+---
+- name: Playbook to demonstrate cv_container module.
+  hosts: cvaas
+  connection: local
+  gather_facts: no
+  collections:
+    - arista.cvp
+
+  tasks:
+    - name: "Gather CVaaS facts from {{ inventory_hostname }}"
+      arista.cvp.cv_facts_v3:
+      register: cvp_facts
+```
+
+If the playbook runs and completes you have confirmed successful communication with CVaaS.
+Adding -vvv to increase verbosity will give you additional information regarding your CVaaS instance when running the cvaas_facts.yml playbook.
+
+Now that AVD is talking to the CVaaS service you can run the "cvaas_deploy.yml" playbook to build out your containers, move devices to the proper container and then apply the generated config to the device.
+
+## Sample playbook cvaas_deploy.yml
+
+```text
+---
+- name: Configlet upload management
+  hosts: cvaas
+  connection: local
+  gather_facts: no
+  tasks:
+
+  - name: "Deploying configurations to CVaaS"
+    import_role:
+      name: arista.avd.eos_config_deploy_cvp
+    vars:
+      container_root: 'DC1_FABRIC'
+      configlets_prefix: 'DC1'
+      state: present
+      cv_collection: v3
+      execute_tasks: false
+```
+
+!!! note
+    For details on the supported inputs for `arista.avd.eos_config_deploy_cvp` see the [role documentation](../README.md)
+
+## Ansible Vault
+
+Once things are working it's a good idea to use Ansible Vault to encrypt your passwords in a production environment.
+
+    ```text
+    ansible-vault encrypt_string '<super long password>' --name 'ansible_ssh_pass' >> my_file.txt
+    ```
+
+## Key points
+
+1. When creating the vault sometimes there will be an extra "%" sign at the end. Remove this.
+
+    $ANSIBLE_VAULT;1.1;AES256
+    31383837323464376439313531333639373431316433636361633239663632663331383264646639
+    3535386333356537643233376630636265653566636531390a663433323033653736653939663861
+    33313466646363643135353065346439326633326138636331333331333338393332653231643930
+    6661353835373731350a303666343334626532313361376361656235323638646264656639653139
+    3437% <------------------Make sure to remove.
```

Comment:
```suggestion
    ```text
    $ANSIBLE_VAULT;1.1;AES256
    31383837323464376439313531333639373431316433636361633239663632663331383264646639
    3535386333356537643233376630636265653566636531390a663433323033653736653939663861
    33313466646363643135353065346439326633326138636331333331333338393332653231643930
    6661353835373731350a303666343334626532313361376361656235323638646264656639653139
    3437% <------------------Make sure to remove.
    ```
```